packages/mcdoc/src/type/index.ts

```diff
@@ -27,14 +26,46 @@ export namespace NumericRange {
 			return false
 		}
 		if (
-			range.max !== undefined && (RangeKind.isLeftExclusive(range.kind)
-				? val <= range.max
-				: val < range.max)
+			range.max !== undefined && (RangeKind.isRightExclusive(range.kind)
+				? val >= range.max
+				: val > range.max)
 		) {
 			return false
 		}
 		return true
 	}
+
+	export function intersect(a: NumericRange, b: NumericRange): NumericRange {
```

Comment:
I don't know if you actually need the `intersect` function btw, I initially interpreted it more like a union (confirmation bias here, as I thought that's the only function that should be needed).

In which case would you expect to range errors for one child to be thrown that need to be intersected? The only reason for two range errors to be about the same child I can think of is if they are different members of a union and maybe need to be combined because they have overlap.

So yeah what's the case the logic would break without this?

Reply:
Yeah while I was inplementing this I couldn't think of any way the intersect is necessary, because we don't have an `&` mcdoc type. But now that we have it I don't see a reason to get rid of it 🤷